docs: [FC-0074] how-tos and quickstart style review (#263)
openedx/openedx-filters: docs/how-tos/create-a-new-filter.rst

Open edX Filters are supported and maintained by the Open edX community. This mechanism is designed to be extensible and flexible to allow developers to create new filters to implement custom behavior in the application. This guide describes how to create a new Open edX filter with long-term support by following the practices outlined in the :doc:`../decisions/0007-filter-design-practices` ADR.

> Filters designed with long-support follow closely the practices described in the ADR to minimize breaking changes and maximize compatibility and support for future versions of Open edX.

Reviewer: ```suggestion
Filters designed with long-support follow closely the practices described in the ADR to minimize breaking changes and maximize compatibility and support for future Open edX versions.
```

For this, choose a specific point in the service where the filter should be triggered. This could be a method in a service, a view, or a model where the logic that you want to modify is executed. The triggering logic should be consistent and narrow to ensure that the filter is triggered only when the conditions are met. For instance, the triggering logic should be a place where all enrollment logic goes through, ensuring that the filter is triggered consistently when a user enrolls in a course. This could be the ``enroll`` method in the enrollment model in the LMS, which is called when a user enrolls in a course in all cases.

.. note:: When designing a filter, consider the support over time of the service and triggering logic. If the service is likely to change or be deprecated, consider the implications of implementing the filter in that service.

> .. note:: It is helpful to inspect the triggering logic to review  the available data when the filter is triggered. This will help you determine the arguments of the filter and how the filter can modify the behavior.

Reviewer: ```suggestion
.. note:: It is helpful to inspect the triggering logic to review the available data when the filter is triggered. This will help you determine the arguments of the filter and how the filter can modify the behavior.
```

As a rule of thumb, start by passing the most relevant context data from the application flow, and then gradually add more details as you analyze the behavior of the triggering logic.

> .. note:: Considering the criticality of the arguments, could they be removed in the near future? This would mean introducing breaking changes to the filter.

Reviewer: ```suggestion
.. note:: Consider the criticality of the arguments. For example, could they be removed in the near future? This would mean introducing breaking changes to the filter.
```
